CHANGELOG.md

```diff
@@ -137,6 +137,8 @@
   do not get an owner relationship set to the `ControlPlane` anymore hence
   they are not deleted when the `ControlPlane` is deleted.
   [#1099](https://github.com/Kong/gateway-operator/pull/1099)
+- Migrate KGO CRDs to the kubernetes-configuration repo.
```

Comment:
This is a breaking change for users that rely on Go API.

We should document this more thoroughly, explaining how to migrate from "old" types (defined in this repo) to "new" ones (defined in kubernetes-configuration).

I think that adding a note with this in the changelog itself is a good approach but I can be persuaded otherwise.

Reply:
Cloud Gateways may be relying on these APIs, right? We could also give them a Slack heads up just in case.